### DISCUSSION TITLE
Install specific modules

### DISCUSSION THREAD

**Discussion by thierionvi** [OSGeo/grass#3190] [Q&A]
Hello,

Firstly, I'd like to thank you for developing the Grass software, which fully meets our operational needs. We are developing a remote sensing processing chain for mapping land cover. It's called iota²: https://framagit.org/iota2-project/iota2
Some of the modules in this chain can be used to vectorise land cover rasters, simplify polygons and smooth them. We chose Grass because it supports a topological vector format and its management of geometry errors is very efficient. We use certain Grass modules to carry out these operations with python :
- g.proj
- g.region
- r.in.gdal
- r.to.vect
- v.out.ogr
- v.in.ogr
- v.generalize
- v.clean

iota² is installed using the CONDA package manager and there are many dependencies. Our question is quite simple: would it be possible to install these modules without having to install the entire grass software ?

Thanks in advance

Vincent

**Comment by wenzeslaus** [comment]
A DIY solution is to delete all parts of GRASS GIS you don't need. Proper testing is necessary, but GRASS GIS will complain only when you try to use something which is missing. GUI would be unhappy, but the individual tools are independent unless they happen to use other tools (typically those implemented in Python do). Another example is packaging on Linux where you can install GRASS GIS without GUI.

Is what you are asking possible now? No. In general? Yes. Exactly as you are asking? Maybe. We don't have an official conda package for GRASS GIS just yet, but that's [work in progress](https://github.com/nilason/grass-conda). Follow up there if you are interested. Another option would be to build only a part from the source code. This might be possible with or added to the new [CMake build in prep](https://github.com/OSGeo/grass/pull/3021), so you can follow up there as well.
